src/test/fuzz/versionbits.cpp

```diff
@@ -281,47 +260,46 @@ FUZZ_TARGET_INIT(versionbits, initialize)
 
     // since is straightforward:
     assert(since % period == 0);
-    assert(0 <= since && since <= current_block->nHeight + 1);
+    assert(0 <= since && since <= height);
     if (state == exp_state) {
         assert(since == exp_since);
     } else {
-        assert(since == current_block->nHeight + 1);
+        assert(since == height);
     }
 
     // state is where everything interesting is
     switch (state) {
     case ThresholdState::DEFINED:
         assert(since == 0);
         assert(exp_state == ThresholdState::DEFINED);
-        assert(current_block->GetMedianTimePast() < checker.m_begin);
-        assert(current_block->GetMedianTimePast() < checker.m_end);
+        assert(never_active_test || height < checker.m_begin);
         break;
     case ThresholdState::STARTED:
-        assert(current_block->GetMedianTimePast() >= checker.m_begin);
-        assert(current_block->GetMedianTimePast() < checker.m_end);
+        assert(height >= checker.m_begin);
         if (exp_state == ThresholdState::STARTED) {
+            assert(height < checker.m_end);
             assert(blocks_sig < threshold);
         } else {
             assert(exp_state == ThresholdState::DEFINED);
         }
         break;
     case ThresholdState::LOCKED_IN:
         assert(exp_state == ThresholdState::STARTED);
-        assert(current_block->GetMedianTimePast() < checker.m_end);
         assert(blocks_sig >= threshold);
         break;
     case ThresholdState::ACTIVE:
         assert(exp_state == ThresholdState::ACTIVE || exp_state == ThresholdState::LOCKED_IN);
         break;
     case ThresholdState::FAILED:
-        assert(current_block->GetMedianTimePast() >= checker.m_end);
-        assert(exp_state != ThresholdState::LOCKED_IN && exp_state != ThresholdState::ACTIVE);
+        assert(height >= checker.m_begin);
+        assert(height >= checker.m_end);
+        assert(exp_state == ThresholdState::FAILED || exp_state == ThresholdState::STARTED);
```

Comment:
In commit "Migrate versionbits to use height instead of MTP":

I believe you should do `assert(exp_state == ThresholdState::FAILED || (exp_state == ThresholdState::STARTED && blocks_sig < threshold));` here instead. As stated, nothing prevents the FSM implementation from going from STARTED->FAILED when the threshold is reached in the last period.

Reply:
Done